src/main/java/com/mbcoder/iot/gpslogger/GPS_Logger.java

```diff
@@ -0,0 +1,344 @@
+
+        // set the title and size of the stage and show it
+        stage.setTitle("GPS Logger");
+        stage.setWidth(800);
+        stage.setHeight(100);
+
+        // create a JavaFX scene with a stack pane as the root node and add it to the scene
+        StackPane stackPane = new StackPane();
+        Scene scene = new Scene(stackPane);
+        stage.setScene(scene);
+        stage.show();
+
+        // Hbox for the buttons
+        HBox hBox = new HBox();
+        stackPane.getChildren().add(hBox);
+
+        // Button to get offline geodatabase
+        Button btnDownloadOfflineGDB = new Button("Get offline gdb");
+        btnDownloadOfflineGDB.setOnAction(event -> downloadOfflineGDB());
+        hBox.getChildren().add(btnDownloadOfflineGDB);
+
+        // Button to open offline geodatabase
+        Button btnOpenGDB = new Button("Open offline gdb");
+        btnOpenGDB.setOnAction(event -> {
+            System.out.println("opening gdb");
+            openGeodatabase();
+        });
+        hBox.getChildren().add(btnOpenGDB);
+
+        // Button to open the serial port and start processing GPS data
+        btnStartGPSUpdate = new Button("start GPS");
+        btnStartGPSUpdate.setOnAction(event -> startGPSUpdate());
+        btnStartGPSUpdate.setDisable(true);
+        hBox.getChildren().add(btnStartGPSUpdate);
+
+        // Button to start recording the GPS position to offline gdb every 10 seconds
+        btnStartLogger = new Button("start logging");
+        btnStartLogger.setOnAction(event -> startLogging());
+        btnStartLogger.setDisable(true);
+        hBox.getChildren().add(btnStartLogger);
+
+        // Button to sync gps positions collected into the hosted feature service
+        btnSync = new Button("Sync data");
+        btnSync.setOnAction(event -> {
+            System.out.println("sync data");
+            syncGeodatabase();
+        });
+        btnSync.setDisable(true);
+        hBox.getChildren().add(btnSync);
+    }
+
+    /**
+     * Method to start logging GPS data into the geodatabase.
+     * <p>
+     * GPS data is reported very frequently, but this method only records new
+     * data at least every 10 seconds to limit the amount of data recorded.
+     */
+    private void startLogging() {
+        loggingTimer = new Timer();
+
+        loggingTimer.schedule( new TimerTask() {
+            public void run() {
+                // log position to db if there is a new gps update
+                System.out.println("logging triggered after 10 seconds");
+
+                if (featureUpdated) {
+                    System.out.println("logging new position");
+                    table.addFeatureAsync(latestPosition);
+
+                    // flag we've read it
+                    featureUpdated = false;
+                }
+
+            }
+        }, 1000, 10000);
+    }
+
+    /**
+     * Method to connect to the serial port with the UDB GPS device and listen for incoming NMEA sentences
```

Comment:
I think this is a typo?
```suggestion
     * Method to connect to the serial port with the USB GPS device and listen for incoming NMEA sentences
```